File: tests/nodejs/cases/case_098__inventory.js
/**
 * Task case 098: _inventory
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case098Inventory(input) {
  // Task: internal inventory records and prepare transport-ready payload.
  const taskName = '_inventory';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'inventory',
    stage: '',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  // ESCAPE: closure captures payload and handler is retained in module-level handlers.
  retainedHandlers.push(() => payload.checksum);
  return 'ok';
}

module.exports = {
  case098Inventory,
};

File: tests/go/cases/case_098__inventory.go
package escape_tests

var retainedCase098 = []map[string]string{}

func Case098Inventory(input string) string {
    // Task case 098:  inventory payload for downstream business processing.
    raw := input
    if raw == "" {
        raw = "sample"
    }
    payload := map[string]string{
        "task": "_inventory",
        "entity": "inventory",
        "stage": "",
        "input": raw,
    }
    // ESCAPE: payload is captured by retained closure metadata.
    signature := payload["task"] + ":" + payload["input"]
    retainedCase098 = append(retainedCase098, map[string]string{"handler": signature})
    return "ok"
}
